bcipy/tasks/rsvp/main_frame.py

```diff
@@ -145,7 +143,7 @@ def __init__(self,
         self.stopping_evaluator = stopping_evaluator
 
         # Stimuli Agent
-        self.stimuli_agent = stimuli_agent
+        self.stimuli_agent = RandomStimuliAgent(alphabet=self.alphabet)
```

Comment:
If the caller does `d = DecisionMaker(stimuli_agent=MyStimuliAgent())` then this will ignore it.

I think it should probably be something like:
```python
self.stimuli_agent = stimuli_agent if stimuli_agent is not None else RandomStimuliAgent(self.alphabet)
```

Reply:
The test at https://github.com/CAMBI-tech/BciPy/blob/1.5.1/bcipy/helpers/tests/test_copy_phrase_wrapper.py#L161 fails due to this issue I believe

Reply:
We should probably be consistent with how we deal with these agents / evaluators. Either accept None and construct as you do for the agent, default as you do with the evaluator or remove the defaults and fail without it as an argument. I'm not a huge fan of all these defaults, but that's a discussion we can have outside this PR / in another ticket.

Reply:
So the original version that Aziz wrote set the default value of that argument to a `RandomStimuliAgent(...)`.
We still want the default to be this way, but if the user provides a different agent, we should use that instead.

What you said here is the key point:
> and constructed in the init method **if it's not provided**

Notice that in the current code, the stimuli agent is ALWAYS created as a `RandomStimuliAgent(...)`, even if the user tries to construct something else.
So one way or another, you'll have to inspect the user input and decide whether to create that default random agent, or to use the provided one.

There are 2 good reasons to construct this default `RandomStimuliAgent(...)` inside the body of the function, and not have it as part of the `def` line.
1. If a mutable object is used as a default in the definition of a function, then _all class instances_ will share the exact same object. They won't have different copies, they'll literally have a pointer to the same object. For example:
```python
class Trouble:
    def __init__(self, contents=[]):
        self.contents = contents

breakfast = Trouble()
dinner = Trouble()
print(breakfast.contents)
dinner.contents.append("escargot")
print(breakfast.contents) # Ew! escargot for breakfast!
```
2. The object being constructed (the agent) has some parameters (alphabet) that were hard-coded, but actually should be kept in sync with the other arguments being passed. The only way to achieve this is by constructing it in the body of the function.

Reply:
looking back at Matt's suggestion, he has an even more slick way to do this inline (https://github.com/CAMBI-tech/BciPy/pull/108#discussion_r539775439):
```python
self.stimuli_agent = stimuli_agent or RandomStimuliAgent(...)
```
This achieves the same effect.

P.S. - The current code also causes the tests to fail - have you run the unit tests with `pytest --mpl -k "not slow"`?

Reply:
@nik-sm Thank you for your explanation! On last week's sprint planning Matt explained the OR usage:
```self.stimuli_agent = stimuli_agent or RandomStimuliAgent(alphabet=self.alphabet)```
That was my bad that I couldn't understand his suggestion. Now I implemented it.

Which errors did you get during the tests? I see couple of `ModuleNotFoundError` while running `pytest --mpl -k "not slow"`. Tests passed for the decision maker: `coverage run --branch --source=bcipy/tasks/rsvp -m pytest -k TestDecisionMaker`.